include/up-cpp/rpc/RpcClient.h

```diff
@@ -19,42 +19,68 @@
  * under the License.
  * 
  * SPDX-FileType: SOURCE
- * SPDX-FileCopyrightText: 2023 General Motors GTO LLC
+ * SPDX-FileCopyrightText: 2024 General Motors GTO LLC
  * SPDX-License-Identifier: Apache-2.0
  */
 #ifndef _RPC_CLIENT_H_
 #define _RPC_CLIENT_H_
 
 #include <future>
-#include <up-cpp/transport/datamodel/UPayload.h>
-#include <up-cpp/transport/datamodel/UAttributes.h>
-#include "up-core-api/uri.pb.h"
+#include <up-cpp/transport/UListener.h>
+#include <up-cpp/transport/datamodel/UMessage.h>
+#include <up-core-api/uattributes.pb.h>
+#include <up-core-api/ustatus.pb.h>
 
-using namespace uprotocol::utransport;
-using namespace uprotocol::v1;
+namespace uprotocol::rpc {
 
-/**
- * RpcClient is an interface used by code generators for uProtocol services defined in proto files such as
- * the core uProtocol services found in https://github.com/eclipse-uprotocol/uprotocol-core-api. the interface 
- * provides a clean contract for all transports to implement to be able to support RPC on their platform. Each
- * platform MUST implement this interface. For more details please refer to
- * https://github.com/eclipse-uprotocol/uprotocol-spec/blob/main/up-l2/README.adoc[RpcClient Specifications]
- */
-class RpcClient 
-{
-    public:
-        /**
-        * Support for RPC method invocation.
-        * @param topic topic of the method to be invoked (i.e. the name of the API we are calling).
-        * @param payload The request message to be sent to the server.
-        * @param attributes Metadata for the method invocation (i.e. priority, timeout, etc.)
-        * @return Returns the CompletableFuture with the result or exception.
-        */
-        virtual std::future<UPayload> invokeMethod(const UUri &topic, 
-                                                   const UPayload &payload, 
-                                                   const UAttributes &attributes) = 0;
-        
-        virtual ~RpcClient() {} 
-};
+    struct RpcResponse {
```

Comment:
I saw that C++17 is being used, so how about using a [`std::variant`](https://en.cppreference.com/w/cpp/utility/variant) which could express that the `RpcResponse` is one or the other?

For example:
```cpp
using RpcResponse = std::variant<uprotocol::utransport::UMessage, uprotocol::v1::UStatus>;
```

Client in uE code could then do one of two things:
1. Use [`get_if()`](https://en.cppreference.com/w/cpp/utility/variant/get_if) in order to check for the one that's been filled
2. Use the [visitor pattern](https://en.cppreference.com/w/cpp/utility/variant/visit) (should probably recommend this route in docs / examples -- it's safer)

If C++23 were on the table (I don't think so...?), then it'd be possible to use C++23's [`std::expected`](https://en.cppreference.com/w/cpp/utility/expected) directly, which better semantically represents what we want of a 'typical' and 'non-typical' value. (side note: was not aware that had been implemented for C++23! so cool. convergent evolution :slightly_smiling_face:)

Reply:
Not everyone will be using C++17 , for example CHM